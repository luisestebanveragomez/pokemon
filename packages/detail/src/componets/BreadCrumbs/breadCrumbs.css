.breadCrumbs__button {
  background: transparent;
  display: flex;
  align-items: center;
  gap: .2rem;
  color: var(--white);
  cursor: pointer;
  text-decoration: none;
}
.breadCrumbs__button i {
  font-size: 1.5rem;
}
