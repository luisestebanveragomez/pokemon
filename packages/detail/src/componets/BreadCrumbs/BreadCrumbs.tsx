import { FC } from "react";
import { Link } from "react-router-dom";

import "./breadCrumbs.css";

interface BreadCrumbsProps {
  name: string;
  url: string;
}
export const BreadCrumbs: FC<BreadCrumbsProps> = ({ name, url }) => {
  return (
    <Link to={url} className="breadCrumbs__button">
      <i className="icon-arrow-left"></i>
      {name}
    </Link>
  );
};
