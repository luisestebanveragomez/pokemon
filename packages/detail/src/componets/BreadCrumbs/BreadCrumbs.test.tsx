import { render } from "@testing-library/react";
import { BreadCrumbs } from "./BreadCrumbs";

jest.mock("react-router-dom", () => ({
  __esModule: true,
  useNavigate: () => (key: string) => key,
  useParams: () => ({
    id: 12,
  }),
  Link: ({ name, url }) => <a href={url}>{name}</a>,
}));
describe("Bread Crumbs Component", () => {
  it("Should Be Render", () => {
    const { container } = render(<BreadCrumbs name={"test"} url={"/"} />);
    expect(container).toMatchSnapshot();
  });
});
