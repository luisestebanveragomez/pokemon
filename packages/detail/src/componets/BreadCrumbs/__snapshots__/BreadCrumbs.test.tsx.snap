// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bread Crumbs Component Should Be Render 1`] = `
<div>
  <a />
</div>
`;
